webapp/components/utils/UniqueSlugForm.spec.js

```diff
@@ -0,0 +1,80 @@
+import UniqueSlugForm from './UniqueSlugForm'
+import Schema from 'async-validator'
+
+let translate
+let apollo
+let currentUser
+
+beforeEach(() => {
+  translate = jest.fn(() => 'Validation error')
+  apollo = {
+    query: jest.fn().mockResolvedValue({ data: { User: [] } }),
+  }
+  currentUser = null
+})
+
+describe('UniqueSlugForm', () => {
+  let validate = object => {
+    const { formSchema } = UniqueSlugForm({ translate, apollo, currentUser })
+    const validator = new Schema(formSchema)
+    return validator.validate(object, { suppressWarning: true }).catch(({ errors }) => {
+      throw new Error(errors[0].message)
+    })
+  }
+
+  describe('regex', () => {
+    describe('non URL-safe characters, e.g. whitespaces', () => {
+      it('rejects', async () => {
+        await expect(validate({ slug: 'uh oh' })).rejects.toThrow('Validation error')
+      })
+    })
+
+    describe('alphanumeric, hyphens or underscores', () => {
+      it('validates', async () => {
+        await expect(validate({ slug: '_all-right_' })).resolves.toBeUndefined()
```

Comment:
this is confusing for me, it resolves, but to undefined?
can't it return a boolean? or it works like if it fails the validations there is a message to display and if not, there is no message?

Reply:
Yes, it's the specific implementation of [async-validator](https://github.com/yiminghe/async-validator) where you call a callback on validation error and do nothing otherwise. Thus, no return value.